### ISSUE TITLE
Completion list should include property names of constraining type when writing object type literals in type argument positions

### ISSUE THREAD

**Issue by jcalz** [microsoft/TypeScript#33302]
## Search Terms

completion list, IntelliSense, keys, members, generic constraint,

## Suggestion with Example

It would be nice to get some IntelliSense help when specifying the members of a constrained object type.   For example, given these types:

```ts
interface Foo {
  apple: string;
  banana: number;
  cherry: boolean;
}

interface SomeGeneric<F extends Foo> {
  foo: F;
}
```

I was kind of expecting to be given some hints about what keys I needed when writing this:

```ts
declare const s: SomeGeneric<{
  apple: "hey";
  banana: 123;
  // <-- no IntelliSense here for anything but "readonly"
}>;
```

Instead, I only seem to get a suggestion for `readonly`.  It would be very helpful if `apple`, `banana`, and `cherry` were part of the completion list.

## Use Cases

I would think this is useful anytime you're specifying a constrained generic parameter, or possibly even extending an existing interface.  Its absence is painful in any case where the generic type has lots of members, as in [this Stack Overflow question](https://stackoverflow.com/questions/57833594/aggregate-many-generics-in-one-record).

## Related Issues

Possibly related to #1506?

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by williamluke4** [comment]
@DanielRosenwasser If someone can point me in the right direction. I don't mind having a go at implementing this

**Comment by DanielRosenwasser** [comment]
@williamluke4 sorry I missed your message. There is currently another community PR that seems like it's on the right track if you're still interested in contributing and want to get a good handle on what the work here involves.

**Comment by DanielRosenwasser** [comment]
Fixed by #43526. Thanks @tiagovtristao!
